tests/test_metrics/test_losses.py

```diff
@@ -45,38 +45,69 @@ def test_cross_entropy_loss():
         loss = build_loss(loss_cfg)
 
     # test ce_loss
-    cls_score = torch.Tensor([[100, -100]])
-    label = torch.Tensor([1]).long()
-    weight = torch.tensor(0.5)
+    cls_score = torch.Tensor([[-1000, 1000], [100, -100]])
+    label = torch.Tensor([0, 1]).long()
+    class_weight = torch.tensor([0.3, 0.7])  # class 0 : 0.3, class 1 : 0.7
+    weight = torch.tensor([0.5, 0.5])
 
+    # test ce_loss without class weight
     loss_cfg = dict(type='CrossEntropyLoss', reduction='mean', loss_weight=1.0)
     loss = build_loss(loss_cfg)
-    assert torch.allclose(loss(cls_score, label), torch.tensor(200.))
+    assert torch.allclose(loss(cls_score, label), torch.tensor(1100.))
     # test ce_loss with weight
     assert torch.allclose(
-        loss(cls_score, label, weight=weight), torch.tensor(100.))
+        loss(cls_score, label, weight=weight), torch.tensor(550.))
+
+    # test ce_loss with class weight
+    loss_cfg = dict(
+        type='CrossEntropyLoss',
+        reduction='mean',
+        loss_weight=1.0,
+        class_weight=class_weight)
+    loss = build_loss(loss_cfg)
+    assert torch.allclose(loss(cls_score, label), torch.tensor(370.))
+    # test ce_loss with weight
+    assert torch.allclose(
+        loss(cls_score, label, weight=weight), torch.tensor(185.))
 
     # test bce_loss
-    cls_score = torch.Tensor([[100, -100], [100, -100]])
-    label = torch.Tensor([[1, 0], [0, 1]])
-    weight = torch.Tensor([0.5, 0.5])
+    cls_score = torch.Tensor([[-200, 100], [500, -1000], [300, -300]])
+    label = torch.Tensor([[1, 0], [0, 1], [1, 0]])
+    weight = torch.Tensor([0.6, 0.4, 0.5])
+    class_weight = torch.tensor([0.1, 0.9])  # class 0: 0.1, class 1: 0.9
 
+    # test bce_loss without class weight
     loss_cfg = dict(
         type='CrossEntropyLoss',
         use_sigmoid=True,
         reduction='mean',
         loss_weight=1.0)
     loss = build_loss(loss_cfg)
-    assert torch.allclose(loss(cls_score, label), torch.tensor(50.))
+    assert torch.allclose(loss(cls_score, label), torch.tensor(300.))
     # test ce_loss with weight
     assert torch.allclose(
-        loss(cls_score, label, weight=weight), torch.tensor(25.))
+        loss(cls_score, label, weight=weight), torch.tensor(130.))
+
+    # test bce_loss with class weight
+    loss_cfg = dict(
+        type='CrossEntropyLoss',
+        use_sigmoid=True,
+        reduction='mean',
+        loss_weight=1.0,
+        class_weight=class_weight)
+    loss = build_loss(loss_cfg)
+    assert torch.allclose(loss(cls_score, label), torch.tensor(176.667))
+    # test bce_loss with weight
+    assert torch.allclose(
+        loss(cls_score, label, weight=weight), torch.tensor(74.333))
 
     # test soft_ce_loss
     cls_score = torch.Tensor([[100, -100]])
     label = torch.Tensor([[1, 0], [0, 1]])
     weight = torch.tensor(0.5)
+    class_weight = torch.tensor(0.25)
```

Comment:
In `soft_cross_entropy`, `cls_score` should share the same shape with `label`, why they're different here.
And according to `label`, there should be two samples, but `weight` only has one item.